hardware_interface/src/actuator.cpp

```diff
@@ -218,6 +218,7 @@ const rclcpp_lifecycle::State & Actuator::get_state() const { return impl_->get_
 
 return_type Actuator::read(const rclcpp::Time & time, const rclcpp::Duration & period)
 {
+  // TODO(destogl): discuss what should be default return value, e.g., "NOT_EXECUTED"
```

Comment:
how about this return value communicating the state transition (if any) ? 
state is ERROR -> ERROR
state is ACTIVE and read was ok -> ACTIVE
state is ACTIVE and read was not ok -> ERROR
?

Reply:
I don't get your comment. The question here is what should be the return value when read or write are called, and the hardware interface is not in INACTIVE or ACTIVE state. Should this return error? But this is not really an error that should be handled. Should it be quietly (not) executed returning "OK"?

Reply:
Inactive read -> inactive

Reply:
For any non active or inactive we could define a new value of SKIP or INVALID. I think SKIP is less drastic and would work better since we probably don't want to flag up errors when read is called on a non-active-inactive components

Reply:
That's exactly what I am thinking too. So we call it "SKIPPED" instead of "NOT_EXECUTED". Is this correct?

Reply:
Yup